kratos/utilities/function_parser_utility.cpp

```diff
@@ -0,0 +1,184 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//  Collaborator:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+#include "tinyexpr/tinyexpr/tinyexpr.h"
+
+// Project includes
+#include "utilities/function_parser_utility.h"
+#include "utilities/string_utilities.h"
+
+namespace Kratos
+{
+
+GenericFunctionUtility::GenericFunctionUtility(
+    const std::string& rFunctionBody,
+    Parameters LocalSystem
+    ) : mFunctionBody(rFunctionBody)
+{
+    // Correcting function from python style to tinyexpr
+    // NOTE: tinyexpr does not support capital letters
+    std::unordered_map<std::string,std::string> aux_replace_letters({{"**","^"},{"X","x0"},{"Y","y0"},{"Z","z0"}});
+    for (auto& r_pair : aux_replace_letters) {
+        mFunctionBody = StringUtilities::ReplaceAllSubstrings(mFunctionBody, r_pair.first, r_pair.second);
+    }
+
+    // Defining namespace
+    mNameSpace.insert(std::pair<std::string, double>("x", 0.0));
```

Comment:
why is this map needed? It was needed for python, but it looks to me that it should not be needed now